diff --git a/MiniZincIDE/mainwindow.cpp b/MiniZincIDE/mainwindow.cpp
index d6b76e9..badce1c 100755
--- a/MiniZincIDE/mainwindow.cpp
+++ b/MiniZincIDE/mainwindow.cpp
@@ -622,15 +622,18 @@ void MainWindow::init(const QString& projectFile)
     bool hadg12lazyfd = false;
     Solver g12mip("G12 MIP","flatzinc","-Glinear","-b mip",true,false);
     bool hadg12mip = false;
-//    Solver gecode("Gecode (bundled)","fzn-gecode","-Ggecode","",true,false);
-//    bool hadgecode = false;
+    Solver gecode("Gecode (bundled)","fzn-gecode","-Ggecode","",true,false);
+    bool hadgecode = false;
+    Solver gecodeGist("Gecode (Gist, bundled)","fzn-gecode-gist","-Ggecode","",true,true);
+    bool hadgecodegist = false;
 
     int nsolvers = settings.beginReadArray("solvers");
     if (nsolvers==0) {
+        solvers.append(gecode);
+        solvers.append(gecodeGist);
         solvers.append(g12fd);
         solvers.append(g12lazyfd);
         solvers.append(g12mip);
-//        solvers.append(gecode);
     } else {
         IDE::instance()->stats.solvers.clear();
         for (int i=0; i<nsolvers; i++) {
@@ -653,10 +656,14 @@ void MainWindow::init(const QString& projectFile)
                     solver = g12mip;
                     hadg12mip = true;
                 }
-//                else if (solver.name=="Gecode (bundled)") {
-//                    solver = gecode;
-//                    hadgecode = true;
-//                }
+                else if (solver.name=="Gecode (bundled)") {
+                    solver = gecode;
+                    hadgecode = true;
+                }
+                else if (solver.name=="Gecode (Gist, bundled)") {
+                    solver = gecodeGist;
+                    hadgecodegist = true;
+                }
             } else {
                 IDE::instance()->stats.solvers.append(solver.name);
             }
@@ -668,8 +675,10 @@ void MainWindow::init(const QString& projectFile)
             solvers.append(g12lazyfd);
         if (!hadg12mip)
             solvers.append(g12mip);
-//        if (!hadgecode)
-//            solvers.append(gecode);
+        if (!hadgecodegist)
+            solvers.push_front(gecodeGist);
+        if (!hadgecode)
+            solvers.push_front(gecode);
     }
     settings.endArray();
     settings.beginGroup("minizinc");
